profiles/sssd/password-auth

```diff
@@ -15,22 +16,25 @@ auth        required                                     pam_deny.so
 
 account     required                                     pam_access.so                                          {include if "with-pamaccess"}
 account     required                                     pam_faillock.so                                        {include if "with-faillock"}
+-account    sufficient                                   pam_systemd_home.so                                    {include if "with-homed"}
 account     required                                     pam_unix.so
 account     sufficient                                   pam_localuser.so                                       {exclude if "with-files-access-provider"}
 account     sufficient                                   pam_usertype.so issystem
 account     [default=bad success=ok user_unknown=ignore] pam_sss.so
 account     required                                     pam_permit.so
 
+-password   sufficient                                   pam_systemd_home.so                                    {include if "with-homed"}
```

Comment:
@poettering Shouldn't this go after pam_pwquality?

Reply:
No, before. homed links directly against libpwq and validates *all* passwords you set with it anyway, i.e. both when you initially create the user, and when you change the password later via PAM.

There's also a per-user setting that allows you to disable the password policy, if you like. I use that a lot when testing (the policy is just too frickin' annoying when you test passwords all the time). Since pam_pwquality can't really support that (as it has no idea of homed user records), it's hence kinda essential that validation is left to homed, which can check that property first, and skip validation if user record says so.

(the password policy ignore flag can only be set by root on the user record, so it's not a security issue).

Anyway, because of that we want to cut the pw change stack short: if pam_systemd_home.so owns the user, nothing else should run.